### PR TITLE
Enable TLS for Tonic Hello World gRPC app

### PR DESCRIPTION
**Pull request by conradgrobler**
# Checklist

- [ ] Pull request affects core Oak functionality (e.g. runtime, SDK, ABI)
  - [ ] I have written tests that cover the code changes.
  - [ ] I have checked that these tests are run by [Cloudbuild](cloudbuild.yaml)
  - [ ] I have updated [documentation](docs/) accordingly.
  - [ ] I have raised an [issue](https://github.com/project-oak/oak/issues) to
        cover any TODOs and/or unfinished work.
- [x] Pull request includes prototype/experimental work that is under
      construction.

### CODE REVIEW

File: Cargo.lock [hunk 1]
```diff
@@ -2513,6 +2602,70 @@ name = "wasi"
...
+dependencies = [
+ "bumpalo 3.2.0 (registry+https://github.com/rust-lang/crates.io-index)",
+ "lazy_static 1.4.0 (registry+https://github.com/rust-lang/crates.io-index)",
+ "log 0.4.8 (registry+https://github.com/rust-lang/crates.io-index)",
+ "proc-macro2 1.0.8 (registry+https://github.com/rust-lang/crates.io-index)",
+ "quote 1.0.2 (registry+https://github.com/rust-lang/crates.io-index)",
+ "syn 1.0.14 (registry+https://github.com/rust-lang/crates.io-index)",
+ "wasm-bindgen-shared 0.2.58 (registry+https://github.com/rust-lang/crates.io-index)",
+]
+
```
Comment (tiziano88) [inline Cargo.lock]: Not a huge issue, but do you have any idea why these deps are now needed?

File: experimental/split_grpc/certs/local.key [hunk 2]
```diff
@@ -0,0 +1,27 @@
+-----BEGIN RSA PRIVATE KEY-----
```
Comment (tiziano88) [inline experimental/split_grpc/certs/local.key]: Could you add a README in this folder to explain why we have these secrets here, and commands to re-generate them if necessary?

File: experimental/split_grpc/client/src/main.rs [hunk 3]
```diff
@@ -27,8 +26,18 @@ pub mod proto {
 #[tokio::main]
 async fn main() -> Result<(), Box<dyn std::error::Error>> {
     env_logger::init();
-
-    let mut client = HelloWorldClient::connect("http://[::1]:50051").await?;
+    let root_cert = tokio::fs::read("experimental/split_grpc/certs/ca.pem").await?;
```
Comment (tiziano88) [inline experimental/split_grpc/client/src/main.rs]: Wow, async file reading ;)

Comment (conradgrobler) [reply experimental/split_grpc/client/src/main.rs]: Indeed. :)

**Pull request merged**